### DISCUSSION TITLE
Documentation for older Keycloak versions

### DISCUSSION THREAD

**Discussion by manologg** [keycloak/keycloak#26706] [Q&A] [answered]
This is something I always have problems with.

When I started working with Keycloak at my current company we had a Keycloak 15 cluster and one of the first things I did was migrating to Keycloak 18.

Trying to find Keycloak 18 documentation was a really bad experience since all the existing documentation under [keycloak.org - Guides](https://www.keycloak.org/guides) refers to the latest version. I still do not know how to do it except for trying to find older versions in the [Wayback Machine](https://archive.org/web/).

Other option I found

**1st question:** How do you all deal with this problem?

My current problem: I want to set up a remote Infinispan cluster and the only guide from Keycloak's side is this [guide](https://www.keycloak.org/high-availability/connect-keycloak-to-external-infinispan) that refers to a feature being currently in preview... no idea what parts were still possible in Keycloak 18 (using Quarkus)

**2nd question:** Does anyone knows how changed is that guide? Could I follow it without problem for Keycloak 18?

**3rd question:** (this is more for keycloak mantainers) Could it be possible to publish new documentation in the Keycloak page without having to overwrite the old one? I mean having old and current documentation.

**Comment by manologg** [comment]
Ok I found a way... I compiled the documentation from the old tag and the old repo and published it using GitHub Pages. Not the best but it does the trick:

https://manologg.github.io/keycloak-documentation-18.0.2

**Comment by manologg** [comment]
But anyway I still wonder: where do I find Keycloak 18 Infinispan related documentation? What I found in the old documentation is Wildfly stuff: https://manologg.github.io/keycloak-documentation-18.0.2/server_installation/index.html#assembly-setting-up-crossdc

Is there an older version (18) of https://www.keycloak.org/high-availability/connect-keycloak-to-external-infinispan ?

**Comment by elkashefehab** [comment]
do you have any link for old versions on github

**Reply by manologg** [reply]
https://github.com/keycloak/keycloak-documentation

**Comment by codespearhead** [comment, marked as answer]
Unless you're a paying customer, you should upgrade to the latest stable version as soon as it becomes available [1].

As for the question itself, you can consult the new Documentation Archive [2]. If you're using an older distribution, you can refer to the old Documentation Archive [3], which can be built by running `./build-product.sh`.

[1] https://github.com/keycloak/keycloak/discussions/29126
[2] https://www.keycloak.org/documentation-archive.html
[3] https://github.com/keycloak/keycloak-documentation/tags

**Reply by jeetparte** [reply]
This does not help to look up older versions of any of the guides (https://www.keycloak.org/guides).
